homeassistant/components/amcrest/config_flow.py

```diff
@@ -0,0 +1,233 @@
+
+    async def _create_entry(self):
+        """Create entry for device."""
+
+        return self.async_create_entry(
+            title=self.device_config[CONF_NAME], data=self.device_config
+        )
+
+
+class AmcrestOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle Amcrest device options."""
+
+    def __init__(self, config_entry):
+        """Initialize Amcrest device options flow."""
+        self.config_entry = config_entry
+        self.options = dict(config_entry.options)
+        self.device = None
+
+    async def async_step_init(self, user_input=None):
+        """Manage the Amcrest device options."""
+        self.device = self.hass.data[DOMAIN][DEVICES][self.config_entry.data[CONF_NAME]]
+        return await self.async_step_configure_stream()
+
+    async def async_step_configure_stream(self, user_input=None):
+        """Manage the Amcrest device options."""
+        if user_input is not None:
+            self.options.update(user_input)
+            return self.async_create_entry(title="", data=self.options)
+
+        return self.async_show_form(
+            step_id="configure_stream",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_STREAM_SOURCE,
+                        default=self.config_entry.options.get(
+                            CONF_STREAM_SOURCE, DEFAULT_STREAM_SOURCE
+                        ),
+                    ): vol.In(STREAM_SOURCE_LIST),
+                    vol.Optional(
+                        CONF_FFMPEG_ARGUMENTS,
+                        default=self.config_entry.options.get(
+                            CONF_FFMPEG_ARGUMENTS, DEFAULT_FFMPEG_ARGUMENTS
+                        ),
+                    ): str,
+                    vol.Optional(
+                        CONF_RESOLUTION,
+                        default=self.config_entry.options.get(
+                            CONF_RESOLUTION, DEFAULT_RESOLUTION
+                        ),
+                    ): vol.All(vol.In(RESOLUTION_LIST)),
+                    vol.Optional(
+                        CONF_AUTHENTICATION,
+                        default=self.config_entry.options.get(
+                            CONF_AUTHENTICATION, DEFAULT_AUTHENTICATION
+                        ),
+                    ): vol.All(vol.In(AUTHENTICATION_LIST)),
+                    vol.Optional(
+                        CONF_CONTROL_LIGHT,
+                        default=self.config_entry.options.get(
+                            CONF_CONTROL_LIGHT, DEFAULT_CONTROL_LIGHT
+                        ),
+                    ): bool,
+                    vol.Optional(
+                        CONF_SCAN_INTERVAL,
+                        default=self.config_entry.options.get(
+                            CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+                        ),
+                    ): int,
+                    vol.Optional(
+                        CONF_BINARY_SENSORS,
```

Comment:
Instead of this being a config option, add all entities but allow user disabling them.

Reply:
not all devices and not all different firmware work the same way. enabling them all would probably take more work for the user to disable them all